perception/image_projection_based_fusion/test/test_geometry.cpp

```diff
@@ -0,0 +1,175 @@
+
+  // Test at Shape::CYLINDER
+  shape.type = 1;
+  vertices.clear();
+
+  image_projection_based_fusion::objectToVertices(pose, shape, vertices);
+
+  EXPECT_FALSE(vertices.empty());
+  EXPECT_NEAR(vertices.at(0).x(), 2.732050807568877293528, 1e-6);
+  EXPECT_NEAR(vertices.at(0).y(), 3.0, 1e-6);
+  EXPECT_NEAR(vertices.at(0).z(), 7.0, 1e-6);
+  EXPECT_NEAR(vertices.at(11).x(), 2.732050807568877293528, 1e-6);
+  EXPECT_NEAR(vertices.at(11).y(), 1.0, 1e-6);
+  EXPECT_NEAR(vertices.at(11).z(), -1.0, 1e-6);
+
+  // Test at Shape::POLYGON (Nothing to do)
+  shape.type = 2;
+  vertices.clear();
+
+  image_projection_based_fusion::objectToVertices(pose, shape, vertices);
+
+  EXPECT_TRUE(vertices.empty());
+}
+
+TEST(transformPoints, test_transformPoints)
+{
+  std::vector<Eigen::Vector3d> input_points;
+  Eigen::Vector3d point(0.0, 0.0, 0.0);
+  input_points.push_back(point);
+  Eigen::Translation<double, 3> translation(1.0, 1.0, 1.0);
+  Eigen::Matrix3d rotation = (Eigen::AngleAxisd(M_PI / 4, Eigen::Vector3d::UnitX()) *
+                              Eigen::AngleAxisd(M_PI / 4, Eigen::Vector3d::UnitY()) *
+                              Eigen::AngleAxisd(M_PI / 4, Eigen::Vector3d::UnitZ()))
+                               .toRotationMatrix();
+  Eigen::Affine3d affine_transform = rotation * translation;
+  std::vector<Eigen::Vector3d> output_points;
+
+  image_projection_based_fusion::transformPoints(input_points, affine_transform, output_points);
+
+  EXPECT_FALSE(output_points.empty());
+  EXPECT_NEAR(output_points.at(0).x(), 0.7071067811865475244008, 1e-6);
+  EXPECT_NEAR(output_points.at(0).y(), 0.5, 1e-6);
+  EXPECT_NEAR(output_points.at(0).z(), 1.5, 1e-6);
+}
+
+TEST(is_inside, test_is_inside)
+{
+  // Test default pattern
+  sensor_msgs::msg::RegionOfInterest outer;
+  outer.x_offset = 30;
+  outer.y_offset = 40;
+  outer.height = 400;
+  outer.width = 300;
+  sensor_msgs::msg::RegionOfInterest inner;
+  inner.x_offset = 31;
+  inner.y_offset = 41;
+  inner.height = 399;
+  inner.width = 299;
+  const double outer_offset_scale = 1.0;
+
+  bool inside_flag = image_projection_based_fusion::is_inside(outer, inner, outer_offset_scale);
+
+  EXPECT_TRUE(inside_flag);
+
+  // Test left-top outside pattern
+  inner.x_offset = 29;
+  inner.y_offset = 39;
+
+  inside_flag = image_projection_based_fusion::is_inside(outer, inner, outer_offset_scale);
+
+  EXPECT_FALSE(inside_flag);
+
+  // Test right-bottom outside pattern
+  inner.x_offset = 31;
+  inner.y_offset = 41;
+  inner.height = 401;
+  inner.width = 301;
+
+  inside_flag = image_projection_based_fusion::is_inside(outer, inner, outer_offset_scale);
+
+  EXPECT_FALSE(inside_flag);
+}
+
+TEST(sanitizeROI, test_sanitizeROI)
+{
+  // Test default pattern
+  sensor_msgs::msg::RegionOfInterest roi;
+  roi.x_offset = 10;
+  roi.y_offset = 20;
+  roi.height = 200;
+  roi.width = 100;
+  int height_ = 400;  // image height
+  int width_ = 300;   // image width
```

Comment:
```suggestion
  int height = 400;  // image height
  int width = 300;   // image width
```

Reply:
fixed in [be2118e](https://github.com/autowarefoundation/autoware.universe/pull/7096/commits/be2118e505d1b4947ae87b0f954d1cf4d5b8e6c0)